Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/ModalInfoView/ModalInfoViewController.swift

```diff
@@ -40,8 +41,49 @@ public final class ModalInfoViewController: BaseViewController, TitledViewContro
             } else {
                 bodyLabel.text = viewModel.body.value
             }
-            bodyLabel.textColor = .gdsGrey
+            bodyLabel.textColor = (viewModel as? ModalInfoButtonsViewModel)?.bodyTextColour ?? .gdsGrey
             bodyLabel.accessibilityIdentifier = "bodyLabel"
         }
     }
+    
+    @IBOutlet private var primaryButton: RoundedButton! {
+        didSet {
+            if viewModel is ModalInfoButtonsViewModel {
```

Comment:
I think you could combine these a little - the logic here would be if it is that `viewModel`, but there is no `buttonViewModel`, it wouldn't hide the button.

```suggestion
if let modalButtonsViewModel = viewModel as? ModalInfoButtonsViewModel,
   let buttonViewModel = modalButtonsViewModel.primaryButtonViewModel {
    primaryButton.setTitle(buttonViewModel.title, for: .normal)
    primaryButton.accessibilityIdentifier = "modal-info-primary-button"
} else {
    primaryButton.isHidden = true
}
```

Same then for the `secondaryButton`

Reply:
Good shout 🙂